dali/operators/image/convolution/gaussian_blur_gpu.h

```diff
@@ -0,0 +1,141 @@
+    }
+    constexpr int nthreads = 1;
+
+    output_desc.resize(1);
+    output_desc[0].type = TypeInfo::Create<Out>();
+    // The shape of data stays untouched
+    output_desc[0].shape = input.shape();
+
+    params_.resize(nsamples);
+    windows_.resize(nsamples);
+    for (auto &win_shape : window_shapes_) {
+      win_shape.resize(nsamples);
+    }
+
+    kmgr_.template Resize<Kernel>(nthreads, nsamples);
+
+    for (int i = 0; i < nsamples; i++) {
+      params_[i] = GetSampleParams<axes>(i, spec_, ws);
+      windows_[i].PrepareWindows(params_[i]);
+    }
+    RepackAsTL<axes>(window_shapes_, params_);
+    RepackAsTL<axes>(windows_tl_, windows_);
+    auto& req = kmgr_.Setup<Kernel>(0, ctx_, processed_shape.to_static<ndim>(), window_shapes_);
+    return true;
+  }
+
+  void RunImpl(workspace_t<GPUBackend>& ws) override {
+    const auto& input = ws.template InputRef<GPUBackend>(0);
+    auto& output = ws.template OutputRef<GPUBackend>(0);
+    output.SetLayout(input.GetLayout());
+
+    auto processed_shape = input.shape();
+    int nsamples = processed_shape.num_samples();
+    // If we are sequence-like, make sure that all sequence elements are compressed to first dim
+    if (is_sequence) {
+      processed_shape = collapse_dims(processed_shape, {{0, dim_desc_.usable_axes_start}});
+    }
+
+    auto static_shape = processed_shape.to_static<ndim>();
+
+    auto in_view_dyn = view<const In>(input);
+    auto out_view_dyn = view<Out>(output);
+
+    // TODO(klecki): Just create it from the move(in_view_dyn.data), processed_shape
+    auto in_view = reshape<ndim>(in_view_dyn, static_shape);
+    auto out_view = reshape<ndim>(out_view_dyn, static_shape);
+
+    kmgr_.Run<Kernel>(0, 0, ctx_, out_view, in_view, windows_tl_);
+  }
+
+ private:
+  OpSpec spec_;
+  int batch_size_ = 0;
+  DimDesc dim_desc_;
+
+  kernels::KernelManager kmgr_;
+  kernels::KernelContext ctx_;
+
+  std::vector<GaussianBlurParams<axes>> params_;
+  std::vector<GaussianWindows<axes>> windows_;
+  std::array<TensorListShape<1>, axes> window_shapes_;
+  std::array<TensorListView<StorageCPU, const float, 1>, axes> windows_tl_;
+};
+
+template <typename Out, typename In>
+std::unique_ptr<OpImplBase<GPUBackend>> GetGaussianBlurGpuImpl(const OpSpec& spec,
+                                                               DimDesc dim_desc) {
+  std::unique_ptr<OpImplBase<GPUBackend>> result;
+  VALUE_SWITCH(dim_desc.usable_axes_count, AXES, GAUSSIAN_BLUR_SUPPORTED_AXES, (
+    VALUE_SWITCH(static_cast<int>(dim_desc.has_channels), HAS_CHANNELS, (0, 1), (
+      VALUE_SWITCH(static_cast<int>(dim_desc.is_sequence), IS_SEQUENCE, (0, 1), (
+        constexpr bool has_ch = HAS_CHANNELS;
+        constexpr bool is_seq = IS_SEQUENCE;
+          result.reset(new GaussianBlurOpGpu<Out, In, AXES, has_ch, is_seq>(spec, dim_desc));
+      ), (DALI_FAIL("Got value different than {0, 1} when converting bool to int."))); // NOLINT
```

Comment:
This error doesn't tell anything. I would print that it is about sequences/channels.

Reply:
I should put `void();` in here or something, unless we get a bit flip I don't think this can happen.

Reply:
So maybe just unexpected internal error in GuassinBlur GPU OP? We would have the line numer anyway and it won't confuse user with bool to int conversion jibba jabba.

Reply:
Removed the error, by the C++ standard, we will always get 0 or 1